cpp/src/join/hash_join.cu

```diff
@@ -61,12 +61,15 @@ namespace {
  * @return The exact size of the output of the join operation
  */
 template <join_kind JoinKind>
-std::size_t compute_join_output_size(table_device_view build_table,
-                                     table_device_view probe_table,
-                                     cudf::detail::multimap_type const& hash_table,
-                                     bool const has_nulls,
-                                     cudf::null_equality const nulls_equal,
-                                     rmm::cuda_stream_view stream)
+std::size_t compute_join_output_size(
+  table_view const& build_table,
+  table_view const& probe_table,
+  std::shared_ptr<cudf::experimental::row::equality::preprocessed_table> preprocessed_build,
+  std::shared_ptr<cudf::experimental::row::equality::preprocessed_table> preprocessed_probe,
```

Comment:
You don't have to use `shared_ptr` to pass the pointers around. Just use `unique_ptr` and pass their reference.
```suggestion
  std::unique_ptr<cudf::experimental::row::equality::preprocessed_table> const& preprocessed_build,
  std::unique_ptr<cudf::experimental::row::equality::preprocessed_table> const& preprocessed_probe,
```